### PR TITLE
implement Hash for Antichain<T> when T is Ord+Hash

### PR DESCRIPTION
**Pull request by danhhz**
This allows for e.g. indexing differential Descriptions by their lower
to stitch them together.

### CODE REVIEW

File: timely/src/progress/frontier.rs [hunk 1]
```diff
@@ -229,6 +229,16 @@ impl<T: Clone> Clone for Antichain<T> {
 
 impl<T: TotalOrder> TotalOrder for Antichain<T> { }
 
+impl<T: Ord+std::hash::Hash> std::hash::Hash for Antichain<T> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        let mut temp = self.elements.iter().collect::<Vec<_>>();
```
Comment (danhhz) [inline timely/src/progress/frontier.rs:234]: this temp Vec is unfortunate, should I be doing something different? maybe impl Hash for MutableAntichain (which requires that T be Ord) instead?

**Review by danhhz — COMMENTED** [review]
@frankmcsherry I double-checked after office hours and it looked like we indeed didn't have this. no hurry at all! we might want something like this eventually, but not soon

File: timely/src/progress/frontier.rs (hunk 1 above)
Comment (frankmcsherry) [reply timely/src/progress/frontier.rs:234]: This seems fine to me, and is in-line with how we do equality. It just means that `Hash` is not super cheap, but that isn't the end of the world imo. A different take is that we should maintain antichains in normal form (sorted, say), which is a different amount of disruptive.

Comment (danhhz) [reply timely/src/progress/frontier.rs:234]: Okay then it works for me, too! It's still much faster than the alternative (repeatedly iterating through a list to call Eq). Just wanted to make sure I wasn't missing something.

**Comment by frankmcsherry** [comment]
Looks good, and thanks!

**Pull request merged**